### DISCUSSION TITLE
isDirty not working as expected after reset

### DISCUSSION THREAD

**Discussion by ryanbuxton523** [react-hook-form/react-hook-form#5640] [Q&A]
**Describe the bug**
isDirty does not work as expected after calling reset with values containing a nested field array.

**To Reproduce**
1. Modify the input.
2. Set the input back to "0".
3. See isDirty is false as expected.
4. Click the reset button.
5. Modify the input.
6. Set the input back to "0".
7. See isDirty is true even though it should be false. Also, dirtyFields is an empty object, which indicates isDirty should be false.
8. Click the getValues button.
9. See the values do not contain an empty array for subValues even though that was part of the values used in reset. Could this be why isDirty is true?

**Codesandbox link (Required)**
https://codesandbox.io/s/sharp-wilson-dk1lu?file=/src/App.js

**Expected behavior**
isDirty should be false when the input is the same as the values used in reset.

**Desktop (please complete the following information):**
- OS: Windows 10
- Browser: Chrome
- Version: 90.0.4430.212

Thanks for this awesome library!

**Comment by bluebill1049** [comment]
<img width="1081" alt="Screen Shot 2021-06-17 at 8 50 46 am" src="https://user-images.githubusercontent.com/10513364/122304778-1d88d100-cf49-11eb-8610-5157201386b3.png">

This is due to form values is no longer deep equal defaultValues.

**Reply by ryanbuxton523** [reply]
Thank you for the response. As a follow up question, why is an empty array not saved in test.subValues after calling reset as shown in the CSB? I expected reset to behave the same way as initially setting the default values. Is there a different recommended approach for persisting empty arrays after reset?

**Reply by bluebill1049** [reply]
because there is no input present on the screen, and defaultValues is actually series of inputs which hook form gets registered, so for an empty array, hook form wouldn't be able to associate with any inputs.
